src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -1058,6 +1119,136 @@ impl Tree {
     }
 }
 
+/// methods for wildcard borrows
+impl<'tcx> Tree {
+    /// analogous to `perform_access`, but we do not know from which exposed reference the access happens.
+    pub fn perform_wildcard_access(
+        &mut self,
+        access_range_and_kind: Option<(AllocRange, AccessKind, diagnostics::AccessCause)>,
+        global: &GlobalState,
+        alloc_id: AllocId, // diagnostics
+        span: Span,        // diagnostics
+    ) -> InterpResult<'tcx> {
+        if let Some((access_range, access_kind, access_cause)) = access_range_and_kind {
+            for (loc_range, loc) in self.locations.iter_mut(access_range.start, access_range.size) {
+                // We can simply iterate over the entire graph in arbitrary order, as the
+                // relatedness of the access for each node can be calculated with only
+                // the tracking information attached to each node.
+                let mut stack = vec![self.root];
+                while let Some(id) = stack.pop() {
+                    let node = self.nodes.get_mut(id).unwrap();
+                    let mut entry = loc.perms.entry(id);
+                    let perm = entry.or_insert(node.default_location_state());
+                    // TODO narrowing based on protector
+                    let protected = global.borrow().protected_tags.contains_key(&node.tag);
+                    let mut entry = loc.wildcard_accesses.entry(id);
+                    let wildcard_access = entry.or_insert(Default::default());
```

Comment:
It's strange to be inserting this here if we anyway always want to error when there is no node.